apps/src/applab/setPropertyDropdown.js

```diff
@@ -356,22 +417,21 @@ export function setPropertyValueSelector() {
  *   types, provides full list of properties across all types.
  */
 export function setPropertyDropdown() {
-  return function (editor) {
+  return function (aceEditor) {
+    var elementType;
     // Note: We depend on "this" being the droplet socket when in block mode,
     // such that parent ends up being the block. In text mode, this.parent
     // ends up being undefined.
-    var param1 = getFirstSetPropertyParam(this.parent, editor);
-    if (!param1) {
-      return fullDropdownOptions;
-    }
-
-    var elementId = stripQuotes(param1);
-    var element = document.querySelector("#divApplab #" + elementId);
-    if (!element) {
-      return fullDropdownOptions;
+    var param1 = getFirstSetPropertyParam(this.parent, aceEditor);
+    if (param1) {
+      let elementId = stripQuotes(param1);
+      let element = document.querySelector("#divApplab #" + elementId);
+      if (element) {
+        elementType = library.getElementType(element, true);
+      }
     }
 
-    return getDropdownProperties(library.getElementType(element));
+    return getDropdownProperties(this.parent, aceEditor, elementType);
```

Comment:
It looks like it's now possible to call `getDropdownProperties` with an undefined `elementType`.  It's not listed as optional, and I don't think there was a path that allowed this before.  Is there something I'm missing that wouldn't allow this to happen, or is it safe to pass undefined here?

Reply:
It is safe and was safe before. Now, it will fallback upon `if (elementType in PROPERTIES)` - the comment block needs to be updated for that param...